crates/evm/evm.rs

```diff
@@ -242,6 +263,59 @@ pub fn evm_state(store: Store) -> EvmState {
     )
 }
 
+/// Calls the eip4788 beacon block root system call contract
+/// As of the Cancun hard-fork, parent_beacon_block_root needs to be present in the block header.
+pub fn beacon_root_contract_call(
+    state: &mut EvmState,
+    header: &BlockHeader,
+    spec_id: SpecId,
+) -> Result<ExecutionResult, EvmError> {
+    let system_caller_address =
+        RevmAddress::from_slice(&hex::decode("fffffffffffffffffffffffffffffffffffffffe").unwrap());
+    let beacon_root_contract_address = RevmTxKind::Call(RevmAddress::from_slice(
+        &hex::decode("000F3df6D732807Ef1319fB7B8bB8522d0Beac02").unwrap(),
+    ));
+    let beacon_root = header
+        .parent_beacon_block_root
+        .expect("No parent beacon block root on block header");
+
+    let tx_env = TxEnv {
+        caller: system_caller_address,
+        transact_to: beacon_root_contract_address,
+        nonce: None,
+        gas_limit: 30_000_000,
+        value: RevmU256::ZERO,
+        data: revm::primitives::Bytes::copy_from_slice(beacon_root.as_bytes()),
+        gas_price: RevmU256::ZERO,
+        chain_id: None,
+        gas_priority_fee: None,
+        access_list: Vec::new(),
+        blob_hashes: Vec::new(),
+        max_fee_per_blob_gas: None,
+    };
```

Comment:
nit: Most of these are default values so we can just use `...Default::default()` to fill them